diff --git a/liblaunch/vproc_internal.h b/liblaunch/vproc_internal.h
index 8fe762c..ef811fe 100644
--- a/liblaunch/vproc_internal.h
+++ b/liblaunch/vproc_internal.h
@@ -36,7 +36,8 @@ typedef pid_t * pid_array_t;
 typedef mach_port_t vproc_mig_t;
 
-#if defined(job_MSG_COUNT) || defined (xpc_domain_MSG_COUNT)
+#if (defined(job_MSG_COUNT) || defined(xpc_domain_MSG_COUNT)) && \
+    !defined(PUREDARWIN_LIBLAUNCH_CLIENT)
 /* HACK */
 #include "core.h"
 #endif
