diff --git a/osfmk/i386/commpage/commpage.c b/osfmk/i386/commpage/commpage.c
--- a/osfmk/i386/commpage/commpage.c
+++ b/osfmk/i386/commpage/commpage.c
@@ -168,7 +168,7 @@ commpage_allocate(
 	if ((kr = mach_make_memory_entry( kernel_map,		// target map
 				    &size,		// size 
 				    kernel_addr,	// offset (address in kernel map)
-				    uperm,	// protections as specified
+				    uperm | VM_PROT_WRITE,	// preserve user rights and kernel writes
 				    &handle,		// this is the object handle we get
 				    NULL )))		// parent_entry (what is this?)
 		panic("cannot make entry for commpage %d", kr);
